### PR TITLE
feat: Enrollment/Events note persistence

### PR DESCRIPTION
**Pull request by luciano-fiandesio**
- Persist notes for events and enrollment for "new" tracker
- Simpliefied notes validation logic to be in sync with current tracker importer
- Add UID generation for notes
- refactor: removed `commentService` as dependency for all the validation hooks
- add unit and integration tests

ref: DHIS2-9083

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/converter/NotesConverterService.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package org.hisp.dhis.tracker.converter;
+
```
Comment (mortenoh) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/converter/NotesConverterService.java:2]: Copyright

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/NoteValidationUtils.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package org.hisp.dhis.tracker.validation.hooks;
+
```
Comment (mortenoh) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/NoteValidationUtils.java:2]: Copyright

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/hooks/EventNoteValidationHookTest.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
+package org.hisp.dhis.tracker.validation.hooks;
+
```
Comment (mortenoh) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/hooks/EventNoteValidationHookTest.java:2]: Copyright

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/DefaultTrackerBundleService.java [hunk 4]
```diff
@@ -318,6 +327,15 @@ private TrackerTypeReport handleEvents( Session session, TrackerBundle bundle )
             Event event = events.get( idx );
 
             ProgramStageInstance programStageInstance = eventConverter.from( bundle.getPreheat(), event );
+            
+            if (!programStageInstance.getComments().isEmpty()) {
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/DefaultTrackerBundleService.java:331]: Codestyle, brackets on new line.

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/converter/NotesConverterServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
+package org.hisp.dhis.tracker.converter;
+
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/converter/NotesConverterServiceTest.java:2]: Add license.

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/converter/NotesConverterService.java (hunk 1 above)
Comment (luciano-fiandesio) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/converter/NotesConverterService.java:2]: Fixed

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/NoteValidationUtils.java (hunk 2 above)
Comment (luciano-fiandesio) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/NoteValidationUtils.java:2]: Fixed

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/hooks/EventNoteValidationHookTest.java (hunk 3 above)
Comment (luciano-fiandesio) [reply dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/hooks/EventNoteValidationHookTest.java:2]: Fixed

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/DefaultTrackerBundleService.java (hunk 4 above)
Comment (luciano-fiandesio) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/DefaultTrackerBundleService.java:331]: Fixed

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/converter/NotesConverterServiceTest.java (hunk 5 above)
Comment (luciano-fiandesio) [reply dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/converter/NotesConverterServiceTest.java:2]: Fixed

**Pull request closed without merging**